sandbox/namespace_linux.go

```diff
@@ -254,6 +255,37 @@ func (n *networkNamespace) SetGatewayIPv6(gw net.IP) error {
 	return err
 }
 
+func (n *networkNamespace) AddStaticRoute(r *types.StaticRoute) error {
+	n.Lock()
+	defer n.Unlock()
+
+	err := programRoute(n.path, r.Destination, r.NextHop)
+	if err == nil {
+		n.sinfo.StaticRoutes = append(n.sinfo.StaticRoutes, r)
+	}
+	return err
+}
+
+func (n *networkNamespace) RemoveStaticRoute(r *types.StaticRoute) error {
+	n.Lock()
```

Comment:
In libnetwork we never lock the code but only the data so you can just move this `Lock/Unlock` inside the `if`